Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -980,21 +980,13 @@ object ICloneable.Clone()
         public unsafe TValue ReadValue<TValue>()
             where TValue : struct
         {
-            var result = default(TValue);
-
             var state = GetOrCreateActionMap().m_State;
-            if (state != null)
-            {
-                var actionStatePtr = &state.actionStates[m_ActionIndexInState];
-                if (actionStatePtr->phase.IsInProgress())
-                {
-                    var controlIndex = actionStatePtr->controlIndex;
-                    if (controlIndex != InputActionState.kInvalidIndex)
-                        result = state.ReadValue<TValue>(actionStatePtr->bindingIndex, controlIndex);
-                }
-            }
+            if (state == null) return default(TValue);
 
-            return result;
+            var actionStatePtr = &state.actionStates[m_ActionIndexInState];
+            return phase.IsInProgress()
```

Comment:
`actionStatePtr->phase`.

Sidenote: These days, with ref vars now being supported, a whole bunch of the direct pointer code I wrote is probably better written as ref vars.

Reply:
👍